### DISCUSSION TITLE
support for OCI (Oracle Cloud storage) uri

### DISCUSSION THREAD

**Discussion by vigneshtripuraj** [delta-io/delta-rs#1289] [Q&A]
Does this package support OCI (Oracle Cloud storage) uri to read delta tables ? I read that it supports only StorageBackends: AWS S3, Azure Data Lake Storage Gen2, Google Cloud Storage (GCS) and local URI.

**Comment by roeap** [comment]
Not so far, unless oracle cloud storage is S3 compatible, then you might be able to get it to work.

That said, we base our object store support on the object_store crate. There are other external packages that also implement the `ObjectStore` trait from object_store. If one of these is available for oracle you should be able to make that work as well.

**Reply by vigneshtripuraj** [reply]
I believe Oracle Cloud storage is compatible with Amazon S3. Are there are steps that will help in getting it to work.

**Reply by roeap** [reply]
Essentially set up AWS as you would usually, and then also set `AWS_ENDPOINT_URL` to work around the AWS url conventions.
